zulip_bots/zulip_bots/bots/lunch/lunch.py

```diff
@@ -0,0 +1,220 @@
+        if msg[0][0] == '@':
+            user = msg[0][1:].replace('**', '')
+            msg  = msg[1:]
+
+        # Find the command
+        if len(msg) == 0:
+            return  # null
+
+        # <user> I want that
+        if msg[-1] == 'that':
+            return copy_user_list(user, message['sender_full_name'])
+
+        # <user|optional> (I) want <item>
+        if msg[0:2] == ['I', 'want']:
+            return add_to_user_list(user, msg[2:])
+
+        if msg[0] == 'wants':
+            return add_to_user_list(user, msg[1:])
+
+        # <user|optional> remove all
+        if msg == ['remove', 'all']:
+            return remove_all(user)
+
+        # <user|optional> remove <item>
+        if msg[0] == 'remove':
+            return remove_from_user_list(user, msg[1:])
+
+        # orders
+        if msg[0] == 'orders':
+            return show_all_orders()
+
+        # <user|optional> order
+        if msg[0] == 'order':
+            return show_order(user)
+
+        # who should <verb>?
+        if msg[0:2] == ['who', 'should']:
+            return random_user(msg[2:])
+
+        if msg[0] == 'help':
+            return '''
+                <user|optional> (I) want(s) <item>
+                Adds <item> to the sender's (or <user>'s if specified) list of items
+
+                <user> I want that
+                Adds every item in <user>'s list of items to the sender's list of items
+
+                <user|optional> remove <item>
+                Removes <item> to the sender's (or <user>'s if specified) list of items
+
+                <user|optional> remove all
+                Removes all items to the sender's (or <user>'s if specified) list of items
+
+                orders
+                Show all orders
+
+                <user|optional> order
+                Shows all items of the sender's (or <user>'s if specified) list of items
+
+                who should <verb>(?)
+                Randomly selects a user from the orders list
+
+                help
+                Shows help (this section)
+            '''
+
+        return "I don't think I can do that. :frowning:"
+
+# User finder
+# Finds if a user exists
+# error is True, return an error message
+# error is False, create the user line
+def find_user(user, error):
+    try:
+        ORDERS[user]
+    except KeyError:
+        if error:
+            return (True, 'Sry, @**' + user + "** isn't hungry. :disappointed:")
+        else:
+            ORDERS[user] = []
+            return (True, 'User created!')
+    else:
+        return (False, 'User exists!')
+
+# Item interpreter
+# Transforms an array of items into a string
+def item_interpreter(items):
+    item_str = ''
+    for item in items:
+        if len(item_str) == 0:
+            item_str += ' ' + item
+        else:
+            if item_str[-1] == ',':
+                item_str += item
+            else:
+                item_str += ' ' + item
+
+    return item_str[1:].split(',')
```

Comment:
I am bit a confused with this one, your comment says that you are `transforming an array of items into a string`, but you call `split` in `item_str` at the end, which returns a list, not a string. And that's not what the comment says. Am I correct? :) Either way, you can probably get rid of these comments too. :)

Reply:
The function works as expected, but it previously used to return a string (however I forgot to edit the comment :P). The comment has been removed now